test/functional/test_framework/util.py

```diff
@@ -559,6 +559,17 @@ def mine_large_block(node, utxos=None):
     node.generate(1)
 
 
+def generate_to_height(node, target_height):
+    """Generates blocks until a given target block height has been reached.
+       To prevent timeouts, only up to 200 blocks are generated per RPC call.
+       Can be used to activate certain soft-forks (e.g. CSV, CLTV)."""
+    current_height = node.getblockcount()
+    while current_height < target_height:
+        nblocks = min(200, target_height - current_height)
+        current_height += len(node.generate(nblocks))
```

Comment:
Clever idea. An additional sanity assertion could be that `length(result) == nblocks`, as it could get in an infinite loop if `generate` is broken and returns nothing.